prow/jira/jira.go

```diff
@@ -259,8 +288,275 @@ func (jc *client) UpdateRemoteLink(id string, link *jira.RemoteLink) error {
 	return nil
 }
 
-func (jc *client) JiraURL() string {
-	return jc.url
+func (jc *client) DeleteRemoteLink(issueID string, linkID int) error {
+	apiEndpoint := fmt.Sprintf("/rest/api/2/issue/%s/remotelink/%d", issueID, linkID)
+	req, err := jc.upstream.NewRequest("DELETE", apiEndpoint, nil)
+	if err != nil {
+		return err
+	}
+
+	// the response should be empty if it is not an error
+	emptyStruct := struct{}{}
+	resp, err := jc.upstream.Do(req, &emptyStruct)
+	// status code 204 is a success for this function...
+	if resp.StatusCode != 204 && err != nil {
+		return JiraError(resp, err)
+	}
+	return nil
+}
+
+// GenericDeleteRemoteLinkViaURL identifies and removes a remote link from an issue
+// the has the provided URL.
+func GenericDeleteRemoteLinkViaURL(jc Client, issueID, url string) error {
+	links, err := jc.GetRemoteLinks(issueID)
+	if err != nil {
+		return err
+	}
+	for _, link := range links {
+		if link.Object.URL == url {
+			return jc.DeleteRemoteLink(issueID, link.ID)
+		}
+	}
+	return fmt.Errorf("could not find remote link on issue with URL `%s`", url)
+}
+
+func (jc *client) DeleteRemoteLinkViaURL(issueID, url string) error {
+	return GenericDeleteRemoteLinkViaURL(jc, issueID, url)
+}
+
+// FindUser returns all users with a field matching the queryParam (ex: email, display name, etc.)
```

Comment:
The godoc needs to be on the interface, IDEs will only show what is there, not what is in the unexported implementation of that interface